File: consumers/typescript/src/consumer.ts
import * as amqp from 'amqplib';

const host = process.env.RABBITMQ_HOST || 'rabbitmq';
const port = process.env.RABBITMQ_PORT || 5672;
const user = process.env.RABBITMQ_USER || 'guest';
const password = process.env.RABBITMQ_PASSWORD || 'guest';
const queue = process.env.RABBITMQ_QUEUE || 'messages';
const logFile = '/logs/ts.log';

async function start() {
  const connection = await amqp.connect(`amqp://${user}:${password}@${host}:${port}`);
  const channel = await connection.createChannel();

  await channel.assertQueue(queue, { durable: true });

  console.log(`[TS Consumer] Waiting for messages on queue: ${queue}`);

  channel.consume(queue, (msg) => {
    if (msg) {
      const message = msg.content.toString();
      const timestamp = new Date().toISOString();
      const logEntry = `[${timestamp}] TS Consumer received: ${message}\n`;
      
      console.log(logEntry);
      
      const fs = require('fs');
      fs.appendFileSync(logFile, logEntry);
      
      channel.ack(msg);
    }
  });
}

start().catch(console.error);

File: consumers/python/consumer.py
import os
import pika
from datetime import datetime

host = os.getenv('RABBITMQ_HOST', 'rabbitmq')
port = int(os.getenv('RABBITMQ_PORT', 5672))
user = os.getenv('RABBITMQ_USER', 'guest')
password = os.getenv('RABBITMQ_PASSWORD', 'guest')
queue = os.getenv('RABBITMQ_QUEUE', 'messages')
log_file = '/logs/python.log'

credentials = pika.PlainCredentials(user, password)
parameters = pika.ConnectionParameters(host=host, port=port, credentials=credentials)

connection = pika.BlockingConnection(parameters)
channel = connection.channel()

channel.queue_declare(queue=queue, durable=True)

print(f"[Python Consumer] Waiting for messages on queue: {queue}")

def callback(ch, method, properties, body):
    message = body.decode()
    timestamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
    log_entry = f"[{timestamp}] Python Consumer received: {message}\n"
    
    print(log_entry)
    
    with open(log_file, 'a') as f:
        f.write(log_entry)
    
    ch.basic_ack(delivery_tag=method.delivery_tag)

channel.basic_qos(prefetch_count=1)
channel.basic_consume(queue=queue, on_message_callback=callback)

channel.start_consuming()
